src/formnetworks/bilinearformnetwork.jl

```diff
@@ -54,7 +55,7 @@ end
 
 function itensor_identity_map(i_pairs::Vector)
   return prod(i_pairs; init=ITensor(one(Bool))) do i_pair
-    return delta(Bool, dag(first(i_pair)), last(i_pair))
+    return denseblocks(delta(last(i_pair), dag(first(i_pair))))
```

Comment:
Actually I merged prematurely. I think this should be:
```suggestion
return denseblocks(delta(Bool, last(i_pair), dag(first(i_pair))))
```
Otherwise, `delta` will make a tensor with element type `Float64`, which will then promote other tensors to double precision, which is an issue on GPU where it is generally better to use single precision. @JoeyT1994 can you make a new PR adding back `Bool`, or did you see some issue with that?

Reply:
@mtfishman This introduces bugs to me that I don't understand. For instance
```
    g = named_grid((4,))
    s1, s2 = siteinds("S=1/2", g), siteinds("S=1/2", g)

    sind1, sind2 = only(s1[(1,)]), only(s2[(1,)])
    ITensorNetworks.itensor_identity_map([sind1=>prime(sind1), sind2=>prime(sind2)])
```

returns 
<img width="976" alt="image" src="https://github.com/user-attachments/assets/5fa4288e-dc11-4e2c-b4a3-6eed88853ac5" />


Whilst 
```
    #More complex constructor 
    g = named_grid((4,))
    s1, s2 = siteinds("S=1/2", g; conserve_qns =true), siteinds("S=1/2", g; conserve_qns =true)

    sind1, sind2 = only(s1[(1,)]), only(s2[(1,)])
    ITensorNetworks.itensor_identity_map([sind1=>prime(sind1), sind2=>prime(sind2)])
    
```


returns 
<img width="872" alt="image" src="https://github.com/user-attachments/assets/37d6a487-e41a-4ee9-a579-9a63a1cecbce" />

Reply:
If there's a way to avoid these bugs and avoid making it `Float64` type I can make that fix in an new PR

Reply:
I see, interesting. An alternative design could be:
```julia
function itensor_identity_map(elt::Type, i_pairs::Vector)
  return prod(i_pairs; init=ITensor(one(elt))) do i_pair
    return denseblocks(delta(elt, last(i_pair), dag(first(i_pair))))
  end
end

itensor_identity_map(i_pairs::Vector) = itensor_identity_map(Float64, i_pairs)
```
Then we can set the element type as needed.

Reply:
(Though it would be nice to fix that bug anyway.)